application/src/test/java/org/togetherjava/tjbot/commands/SlashCommandAdapterTest.java

```diff
@@ -67,30 +63,26 @@ void getData() {
 
     @Test
     void generateComponentId() {
-        String[] elements = {"foo", "bar", "baz"};
+        // Test that the adapter uses the given generator
         SlashCommandAdapter adapter = createAdapter();
+        adapter.acceptComponentIdGenerator((componentId, lifespan) -> "%s;%s;%s"
+            .formatted(componentId.commandName(), componentId.elements().size(), lifespan));
 
-        String componentIdText = adapter.generateComponentId(elements);
-        ComponentId componentId = assertDoesNotThrow(() -> ComponentIds.parse(componentIdText),
-                "generated component id seems to be invalid with the parser");
-
-        assertEquals(NAME, componentId.getCommandName(),
-                "expected command name to be part of the component id for routing");
-        assertEquals(Arrays.asList(elements), componentId.getElements(),
-                "expected all arguments to carry over the id");
-
-        // Empty elements
-        assertTrue(assertDoesNotThrow(() -> ComponentIds.parse(adapter.generateComponentId()),
-                "component id generation seems to have issues with empty elements").getElements()
-                    .isEmpty());
-
-        // Check that IDs are unique
-        Collection<Integer> ids = new HashSet<>();
-        for (int i = 0; i < UNIQUE_ID_ITERATIONS; i++) {
-            int id = assertDoesNotThrow(() -> ComponentIds.parse(adapter.generateComponentId()),
-                    "generated component id seems to be invalid with the parser").getId();
-            assertFalse(ids.contains(id), "id generator is supposed to create unique IDs");
-            ids.add(id);
+        // No lifespan given
+        String[] elements = {"foo", "bar", "baz"};
+        String[] componentIdText = adapter.generateComponentId(elements).split(";");
+        assertEquals(3, componentIdText.length);
+        assertEquals(NAME, componentIdText[0]);
+        assertEquals(Integer.toString(elements.length), componentIdText[1]);
+        assertEquals(Lifespan.REGULAR.toString(), componentIdText[2]);
+
+        // Explicit lifespan
+        for (Lifespan lifespan : Lifespan.values()) {
+            componentIdText = adapter.generateComponentId(lifespan, elements).split(";");
+            assertEquals(3, componentIdText.length);
+            assertEquals(NAME, componentIdText[0]);
+            assertEquals(Integer.toString(elements.length), componentIdText[1]);
+            assertEquals(lifespan.toString(), componentIdText[2]);
```

Comment:
This can probably be in a separate test, with a better name describing tested behaviour (maybe with a `@DisplayName` too).

But I see other methods are tested the same way, so It's fine i guess. It's general enough, just a bit bulky with lots of assertions. ^^

Reply:
Yeah. I feel like we havent yet "agreed" on some sort of guideline on how to structure tests. Right now we just went with one test-method per method to test and then used comments to separate blocks.

I would say we let this pass through but discuss that topic in a separate GH issue, align on something and then adjust all existing tests (since its not only this file which is affected)

Reply:
Yup, I agree, let's keep it consistent with the current testing practices and revise it later.